geth/api/backend.go

```diff
@@ -208,14 +208,46 @@ func (b *StatusBackend) SendTransaction(ctx context.Context, args common.SendTxA
 	return rst.Hash, nil
 }
 
+func (b *StatusBackend) getVerifiedAccount(password string) (*account.SelectedExtKey, error) {
+	selectedAccount, err := b.accountManager.SelectedAccount()
+	if err != nil {
+		b.log.Error("failed to get a selected account", "err", err)
+		return nil, err
+	}
+	config, err := b.NodeManager().NodeConfig()
+	if err != nil {
+		return nil, err
+	}
+	_, err = b.accountManager.VerifyAccountPassword(config.KeyStoreDir, selectedAccount.Address.String(), password)
+	if err != nil {
+		b.log.Error("failed to verify account", "account", selectedAccount.Address.String(), "error", err)
+		return nil, err
+	}
+	return selectedAccount, nil
+}
+
 // CompleteTransaction instructs backend to complete sending of a given transaction
-func (b *StatusBackend) CompleteTransaction(id common.QueuedTxID, password string) (gethcommon.Hash, error) {
-	return b.txQueueManager.CompleteTransaction(id, password)
+func (b *StatusBackend) CompleteTransaction(id common.QueuedTxID, password string) (hash gethcommon.Hash, err error) {
+	selectedAccount, err := b.getVerifiedAccount(password)
+	if err != nil {
+		_ = b.txQueueManager.NotifyErrored(id, err)
+		return hash, err
+	}
+
+	return b.txQueueManager.CompleteTransaction(id, selectedAccount)
 }
 
 // CompleteTransactions instructs backend to complete sending of multiple transactions
 func (b *StatusBackend) CompleteTransactions(ids []common.QueuedTxID, password string) map[common.QueuedTxID]common.TransactionResult {
-	return b.txQueueManager.CompleteTransactions(ids, password)
+	results := make(map[common.QueuedTxID]common.TransactionResult)
```

Comment:
That isn't necessary, but if you want to store only a result *either* an error you can use interfaces like that: https://play.golang.org/p/-ANFxqKwPsM

Reply:
Nice, i always forget about type switches. I guess i have the preconceived  idea they don't have a good performance :/
I'll try to use this approach next time :)